includes/wikia/api/RevisionUpvotesApiController.class.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+class RevisionUpvotesApiController extends WikiaApiController {
+
+	/**
+	 * Add upvote for given revision made by current user
+	 *
+	 * @requestParam int revisionId
+	 *
+	 * @throws MissingParameterApiException
+	 * @throws NotFoundException
+	 * @throws BadRequestApiException
+	 */
+	public function addUpvote() {
+		$request = $this->getRequest();
+
+		if ( !$request->wasPosted() ) {
```

Comment:
What do you think of for the two write methods, replacing this and the `validateUser` call with instead so you're using the available method for validating the post and token already?
```php
if ( !$user->isLoggedIn() ) {
	throw new UnauthorizedException();
}
$this->checkWriteRequest();
```